test/integration/com/unifina/service/TokenAuthenticatorIntegrationSpec.groovy

```diff
@@ -1,14 +1,13 @@
-package com.unifina.security
+package com.unifina.service
 
 import com.unifina.domain.security.Key
 import com.unifina.domain.security.SecUser
-import grails.test.mixin.Mock
-import spock.lang.Specification
+import com.unifina.security.TokenAuthenticator
+import grails.test.spock.IntegrationSpec
 
 import javax.servlet.http.HttpServletRequest
 
-@Mock([Key])
-class TokenAuthenticatorSpec extends Specification {
+class TokenAuthenticatorIntegrationSpec extends IntegrationSpec {
```

Comment:
I would: 
- leave this as a unit test
- mock `SessionService` (for examples, see subclasses of `BeanMockingSpecification`)
- add a test for the `Bearer` case